tests/intentManager.js

```diff
@@ -57,6 +59,67 @@ define([
             });
         });
 
+        describe("template selector", function() {
+            var vellum,
+                mug,
+                templates =  [
+                    {
+                        name: "Area Mapper",
+                        id: "com.richard.lu.areamapper",
+                        extra: {ext: "value"},
+                        response: {
+                            r1: "x",
+                            r2: "y",
+                            r3: "z",
+                            r4: "",
+                        },
+                    },
+                    {
+                        name: "Barcode Scanner",
+                        id: "com.google.zxing.client.android.SCAN",
+                        extra: {},
+                        response: {},
+                    },
+                    {
+                        name: "Breath Counter",
+                        id: "org.commcare.respiratory.BREATHCOUNT",
+                    },
+                ];
+            before(function(done) {
+                util.init({
+                    intents: {templates: templates},
+                    core: {onReady: function () {
+                        vellum = this;
+                        mug = util.addQuestion("AndroidIntent", "intent");
+                        util.clickQuestion("intent");
+                        done();
+                    }}
+                });
+            });
+
+            it("should select empty template by default", function () {
+                assert.equal($("[name=property-androidIntentAppId]").val(), "");
+                assert.equal($("[name=property-androidIntentAppId-template]").val(), "");
+            });
+
+            _.each(templates, function (temp) {
+                it("should have " + temp.name + " template", function () {
+                    $("[name=property-androidIntentAppId-template]").val(temp.id).change();
+                    assert.equal($("[name=property-androidIntentAppId]").val(), temp.id);
+                    if (temp.extra) {
+                        var extra = widgets.util.getWidget(
+                                $("[name=property-androidIntentExtra]"), vellum);
+                        assert.deepEqual(extra.getValue(), temp.extra);
+                    }
+                    if (temp.response) {
+                        var response = widgets.util.getWidget(
+                                $("[name=property-androidIntentResponse]"), vellum);
+                        assert.deepEqual(response.getValue(), temp.response);
+                    }
+                });
+            });
+        });
+
```

Comment:
Oh I meant more like load a form with an appid that wasn't recognized, but I think that is covered by other tests in this file